Makefile.am

```diff
@@ -526,13 +526,13 @@ noinst_TEXINFOS = com_err/et/com_err.texinfo
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libcyrus_min.pc libcyrus.pc libcyrus_sieve.pc libcyrus_imap.pc
 
+# 	com_err/et/init_et.c  (XXX example only? never used?)
 com_err_et_libcyrus_com_err_la_SOURCES = \
 	com_err/et/com_err.c \
 	com_err/et/com_err.h \
 	com_err/et/error_message.c \
 	com_err/et/error_table.h \
 	com_err/et/et_name.c \
-	com_err/et/init_et.c \
```

Comment:
I think the file is indeed not necessary and can be removed both from Makefile.am and from git-tracking.

The problem is, that Cyrus Imap gets bloated over time with unused stuff, which unused stuff makes understanding the whole code harder, apart from creating bigger binaries.

Reply:
Yes, the actual problem is there's no prototype for the function defined in `init_err.c`, thus it generates a warning.  There's no header file for it either, so no place to put a prototype.  Since it's not used elsewhere it's simplest to just ditch it.

I don't think the `com_err` stuff is in any way fragile -- it's just poorly documented and difficult for novices to understand.  It's also a horrid mess of ancient backwards compatibility junk that I've tried to clean a bit of.

Reply:
The automake/autoconf/etc handling of com_err is fragile.  com_err itself, I think we mostly treat as a static blob and don't fiddle with it.

Can you please paste the exact warning it generates.  Thanks

Reply:
Please enable `-Wall -Wextra -Wimplicit -Wreturn-type -Wswitch -Wcomment -Wshadow -Wstrict-prototypes -Wcast-align -Wchar-subscripts -Wmissing-declarations -Wmissing-prototypes -Wno-long-long -Wformat-extra-args -Wundef -Wbad-function-cast -Wwrite-strings -Wdeclaration-after-statement -Wno-sign-conversion` and see for yourself.

In particular for this problem you'll want `Wstrict-prototypes Wmissing-declarations -Wmissing-prototypes`